src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Runtime/PreferStreamAsyncMemoryOverloads.cs

```diff
@@ -0,0 +1,304 @@
+            if (undesiredWriteAsyncMethod == null)
+            {
+                return;
+            }
+
+            IMethodSymbol? undesiredReadAsyncMethodWithCancellationToken = readAsyncMethodGroup.GetFirstOrDefaultMemberWithParameterInfos(undesiredParametersWithCancellationToken);
+            if (undesiredReadAsyncMethodWithCancellationToken == null)
+            {
+                return;
+            }
+
+            IMethodSymbol? undesiredWriteAsyncMethodWithCancellationToken = writeAsyncMethodGroup.GetFirstOrDefaultMemberWithParameterInfos(undesiredParametersWithCancellationToken);
+            if (undesiredWriteAsyncMethodWithCancellationToken == null)
+            {
+                return;
+            }
+
+            // Create the arrays with the exact parameter order of the undesired methods
+            var preferredReadAsyncParameters = new[]
+            {
+                ParameterInfo.GetParameterInfo(readOnlyMemoryType),  // ReadOnlyMemory<byte> buffer
+                ParameterInfo.GetParameterInfo(cancellationTokenType), // CancellationToken
+            };
+
+            var preferredWriteAsyncParameters = new[]
+            {
+                ParameterInfo.GetParameterInfo(memoryType),  // ReadOnlyMemory<byte> buffer
+                ParameterInfo.GetParameterInfo(cancellationTokenType), // CancellationToken
+            };
+
+            // Retrieve the preferred methods, which are used for constructing the rule message
+            IMethodSymbol? preferredReadAsyncMethod = readAsyncMethodGroup.FirstOrDefault(x =>
+                x.Parameters.Count() == 2 && x.Parameters[0].Type is INamedTypeSymbol type && type.ConstructedFrom.Equals(memoryType));
+            if (preferredReadAsyncMethod == null)
+            {
+                return;
+            }
+
+            IMethodSymbol? preferredWriteAsyncMethod = writeAsyncMethodGroup.FirstOrDefault(x =>
+                x.Parameters.Count() == 2 && x.Parameters[0].Type is INamedTypeSymbol type && type.ConstructedFrom.Equals(readOnlyMemoryType));
+            if (preferredWriteAsyncMethod == null)
+            {
+                return;
+            }
+
+            context.RegisterOperationAction(context =>
+            {
+                IInvocationOperation invocation = (IInvocationOperation)context.Operation;
+
+                IOperation parentOperation = invocation.Parent;
+                if (parentOperation == null)
+                {
+                    return;
+                }
+                // Only accept these two cases:
+                // - await {WriteAsync()|ReadAsync()}
+                // - await {WriteAsync()|ReadAsync()}.ConfigureAwait()
+                if (parentOperation.Kind == OperationKind.Await ||
+                        (parentOperation.Parent != null && parentOperation.Parent.Kind == OperationKind.Await &&
```

Comment:
This check is redundant: `parentOperation.Parent != null && parentOperation.Parent.Kind == OperationKind.Await` given your next check is `parentOperation.Parent is IAwaitOperation awaitOperation`